### DISCUSSION TITLE
Repeating Linear Gradient in UnoCSS

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [unocss/unocss#2316] [Q&A] [answered]
I'm trying to migrate my old WindiCSS setup to UnoCSS. I had the following data in my WindiCSS theme:

```js
export default defineConfig({
  theme: {
    extend: {
      backgroundImage: {
        dStripes: 'repeating-linear-gradient(112deg,transparent,transparent 15px,rgba(0,0,0,0.1) 0,rgba(0,0,0,0.1) 30px)',
        lStripes: 'repeating-linear-gradient(112deg,transparent,transparent 15px,rgba(255,255,255,0.3) 0,rgba(255,255,255,0.3) 30px)'
      }
    }
  }
})
```

How exactly can I convert that to UnoCSS? By trial and errors, I landed till [here](https://uno.antfu.me/play/?html=DwEwlgbgUABDCuBaARgcwLwCIAuBTAhgDaIAsADGTKgE77i4B22iA2gIxsBMIuqAulVr0miargAOBbGAapEhGQWqZYCRAAssAVhVwkAdyycymAHxRgAenARTQA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQqAB6RYKDAEM2AG3hpMOfIWBEAFAnVxylGjCoAuOAG1HT5wPpGZnZOHntfPzgRLAA1A2M2VGQGJlYOVE90eJoSCL9%2BTC1PBQBaOVzIpwLgLSSAeSxjbk8mRIrKqlSlGBaoNryxAEp2qpchEXDKlAMoAGtipWMDVXK8pwA3GeADFmNUAAVKQuK2MpHxYd8AXXaYAAtUEAzEPKUIY2gPF6mYVHjPBxTJwAFgADKDigBiUEAVhYyBYBlWUzE5ycTAMWCoYBmqCwPXkGKxOMo%2BI0kTEvkpQyAA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g), but I can't figure out:

1. Either how to add all those values in attributify mode
2. OR where in my UnoCSS config I can add it like the WindiCSS config, so I can simply use `u-bg="dStripes"`.

Either of the above two alternatives is fine by me, but it's a bit difficult to use the code as documentation 😅. I was referring to [this](https://github.com/unocss/unocss/blob/main/packages/preset-wind/src/rules/background.ts), but I don't see any value used from theme like I was shown in: https://github.com/unocss/unocss/discussions/2305.

**Comment by hrishikesh-k** [comment]
I have currently landed on this [attempt](https://uno.antfu.me/play/?html=DwEwlgbgUABDCuBaARgcwLwCIAuBTAhgDaIAsADGTKgE77i4B22iA2gIxsBMIuqAulVr0miargAOBbGAapEhGQWqC6YRswDO2APbiNrbLQYbx%2BMUwA0h-MdPnsAfTYBWcQA8L1VMnwAKMhYBQQB0bACUDgFePv6BcWShEQDMZO58mLAIiAAWWM4ZcEgA7licZJgAfFDAAPTgEBVAA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQqAB6RYKDAEM2AG3hpMOfIWBEAFAnVxylGjCoAuOAG1HT5wPpGZnZOHntfPzgRLAA1A2M2VGQGJlYOVE90eJoSCL9%2BTC1PBQBaOVzIpwLgLSSAeSxjbk8mRIrKqlSlGBaoNryxAEp2qpchEXDKlAMoAGtipWMDVXK8pwA3GeADFmNUAAVKQuK2MpHxYd8AXXaYAAtUEAzEPKUIY2gPF6mYVHjPBxTJwAFgADKDigBiUEAVhYyBYBlWUzEA18qKGQA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g), but it doesn't work and gives a really bad HTML. Thus, a configuration value would help!

**Comment by hrishikesh-k** [comment]
This [attempt](https://uno.antfu.me/play/?html=DwEwlgbgUABDCuBaARgcwLwCIAuBTAhgDaIAsADGTKgE77i4B22iA2gIxsBMIuqANNloMAzgAd81RtgFCxEqQH02AVlEAPPtVTJ8ACjJ8DRgHRsAlAoNad%2Bw3bKmLAZjLqAulVr0miSaILYYAyoiIRBBNSYsAiIABZYylFwSADuWJxkmAB8UMAA9OAQWUA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQqAB6RYKDAEM2AG3hpMOfIWBEAFAnVxylGjCoAuOAG1HT5wPpGZnZOHntfPzgRLAA1A2M2VGQGJlYOVE90eJoSCL9%2BTC1PBQBaOVzIpwLgLSSAeSxjbk8mRIrKqlSlGBaoNryxAEp2qpchEXDKlAMoAGtipWMDVXK8pwA3GeADFmNUAAVKQuK2MpHxYd8AXXaYAAtUEAzEPKUIY2gPF6mYVHjPBxTJwAFgADKDigBiUEAVhYyBYBlWUzEA18qKGQA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g) finally worked, but please let me know if I can add this to configuration instead of HTML.

**Comment by hrishikesh-k** [comment]
Also, am I right to notice that, the only way to pass custom background size is by using `[length:]`? I found it [here](https://github.com/unocss/unocss/blob/main/packages/preset-wind/src/rules/background.ts#L57). So, `background-size: 300%` had to be `[length:300%]`.

That was really confusing.

**Comment by zyyv** [comment, marked as answer]
> OR where in my UnoCSS config I can add it like the WindiCSS config, so I can simply use u-bg="dStripes"

Hi, you can move your written styles to `shortcuts` [Playground](https://uno.antfu.me/play/?html=DwEwlgbgUABDCuBaARgcwLwCIAuBTAhgDaIAsADGTKgE77i4B22iA2gIxsBMIuqAulVr0miargAOBbGAapEhGQWqC6YRswDO2APbiNrbLQYbx%2BMUwA0h-MdPnsAfTYBWcQA8L1VMnwAKMhYBQQB0bACUDgFePv6BcWShEQDMZO58mLAIiAAWWM4ZcEgA7licZJgAfFDAAPTgEFXV9ZlIaFggAMqGYJIaBVm5mPktiCWYZZXVdZAVQA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQqAB6RYKDAEM2AG3hpMOfIWBEAFAnVw4VABbQYSjlQBcDp07kWIgBaZABlJmAwajlveSDg20pkYIAWAAZ0uCIoA2RgVCwYYIBtAEYygCY0IgBdbNz8wuLKaINOLBDjbFQDKAa8gqLgqhgIMCpSpgMsKjA%2B5pJp2fnKIoB9MoBWMC0SKCIWA1t0klPzgDoygEp104Ojk7Pn9KvbgGZ03drrjScxEiOciUGgwHxwEpApz8UEMSLsTg8exQ-wiLAANQMxjYqGQcOYCNQcXQWJogP8-n4mC0cQUwTk5IpwIwwC0uIA8lhjNw4kwcYyKaNmEoYLyoPyUWJrgLmaDhBBkUyUH0ANa0pTGAyqBkopwANz6wAMLGMqAACpRqbS2PSZVLGbVGTAXKgQETECilBBjNBwX4mTBesY4v6lRl0rSAMTpLYsZBHHVKsSSoHJqVAA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

**Reply by hrishikesh-k** [reply]
I didn't know shortcuts can be used for attributify as well! Thanks!
